pkg/controller/direct/directbase/operations.go

```diff
@@ -57,3 +85,106 @@ func NewDeleteOperation(object *unstructured.Unstructured) *DeleteOperation {
 func (o *DeleteOperation) GetUnstructured() *unstructured.Unstructured {
 	return o.object
 }
+
+func (o *operationBase) UpdateStatus(ctx context.Context, typedStatus any, readyCondition *v1alpha1.Condition) error {
+	status, err := runtime.DefaultUnstructuredConverter.ToUnstructured(typedStatus)
+	if err != nil {
+		return fmt.Errorf("error converting status to unstructured: %w", err)
+	}
+
+	old, _, _ := unstructured.NestedMap(o.object.Object, "status")
+	if old != nil {
+		if status["conditions"] == nil {
+			status["conditions"] = old["conditions"]
+		}
+		// status["observedGeneration"] = old["observedGeneration"]
+		if status["externalRef"] == nil {
+			status["externalRef"] = old["externalRef"]
+		}
```

Comment:
Mind adding the condition that if not equal, return errors as well? I'm interested about the externalRef especially for the `DataflowFlexTemplateJob` since it looks like the JobID has some specialties.

Reply:
This externalRef will change, because it includes the job id :-(

That's one reason why I think we might currently have the wrong primary key for this resource ... trying to track that down internally.